### PR TITLE
Prompt Security's extension integration 

### PR DESCRIPTION
**Pull request by guy-ps**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

> In this PR, I added the functionality that will allow this vscode extension and Prompt Security's functionally in the browser extension communicate.

The process:

- The vscode extension creates a local express server that awaits API calls from the browser extension
- Listens to text selection events and saves the last 25
- The browser extension calls the local server if code is sent as part of the prompt
- If the code matches a selected section the vscode extension returns the code and the file
- The browser extension shows a pop up message

Things left to do:

- [ ] Create a checkbox to activate Prompt Security's part of the extension
- [ ] Consume the local server's port number dynamically  

### References

> Include supporting link to GitHub Issue/PR number

### Testing

> The Prompt security vscode extension was tested in house

I've been having issues with the as-cli-javascript-wrapper package, so I wasn't able to test it post integration.

### Checklist

- [X] I have added documentation for new/changed functionality in this PR (if applicable).
- [?] All active GitHub checks for tests, formatting, and security are passing
- [X] The correct base branch is being used

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -203,8 +209,36 @@ export async function activate(context: vscode.ExtensionContext) {
     kicsScanCommand.registerKicsRemediation();
     // Refresh sca tree with start scan message
     scaResultsProvider.refreshData(constants.scaStartScan);
+    
+    //You will need to add a checkbox to enable the extension's activation
+    const isPromptEnabled:boolean = false;
+    //You will need to change it to recieve a port from the user
+    const port:number = 3312;
```
Comment (OrShamirCM) [inline src/extension.ts:216]: please extract it to a private method ("registerPromptListener" ?)

Comment (guy-ps) [reply src/extension.ts]: Added it, and committed it now, let me know if it wasn't what you meant, or you need it to have some different functionality.

Comment (OrShamirCM) [reply src/extension.ts]: What I meant is about ine 212 to 235, all the logic. that we will have there only one line for registration

Comment (guy-ps) [reply src/extension.ts]: Hi @OrShamirCM and @tiagobcx , can you please review the changes I made on Sunday?

File: src/utils/listener/promptSecurity.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        });
+        //If the window is changed - Prompt
+        vscode.window.onDidChangeWindowState(async windowState => {
+            if (windowState){
+            	this.windowChange();
+            }
+        });
+	}
+	extensionListener(
+	){
```
Comment (OrShamirCM) [inline src/utils/listener/promptSecurity.ts:48]: add try catch pleas

File: src/utils/listener/promptSecurity.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+		this.app.get('/api/health', (req,res) => {
+			res.json({ "vscode":"Yay!" });
+		});
+		this.app.post('/api/checkCode', (req, res) => {
+			const receivedText: string = req.body.text;
```
Comment (OrShamirCM) [inline src/utils/listener/promptSecurity.ts:53]: please check for null

File: src/utils/listener/promptSecurity.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	selectionBuffer(
+		event:vscode.TextEditorSelectionChangeEvent
+	){
+		if (event.selections.length > 0) {
+            const selection = event.selections[0];
+            if (!selection.isEmpty) {
+                const editor = event.textEditor;
+                const code  = editor.document.getText(selection).trim();
+				if (code.length > 20 && code.length < 32* 1024) {
+					const filePath = editor.document.uri.fsPath;
```
Comment (OrShamirCM) [inline src/utils/listener/promptSecurity.ts:83]: check for null please (uri / docment/ editor)

**Review by OrShamirCM — COMMENTED** [review]
Can you add unit test please?

File: src/utils/listener/promptSecurity.ts (hunk 3 above)
Comment (guy-ps) [reply src/utils/listener/promptSecurity.ts]: Added

File: src/utils/listener/promptSecurity.ts (hunk 2 above)
Comment (guy-ps) [reply src/utils/listener/promptSecurity.ts:53]: Added

File: src/utils/listener/promptSecurity.ts (hunk 4 above)
Comment (guy-ps) [reply src/utils/listener/promptSecurity.ts]: Added

File: src/utils/listener/promptSecurity.ts [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+	getServer(){
+		return this.server;
+	}
+	deactivate(){
+		if (this.server){
+			this.server.close();
+		}
+	}
+	registerPromptListener(){
+		try {
```
Comment (OrShamirCM) [inline src/utils/listener/promptSecurity.ts:31]: please make sure al l methods have try-catch

Comment (guy-ps) [reply src/utils/listener/promptSecurity.ts:39]: Done, I also added try/catch to the PromptSecurity register in the extension.ts file

**Pull request closed without merging**